test-data/unit/check-classes.test

```diff
@@ -7134,3 +7134,111 @@ class B(A):  # E: Final class __main__.B has abstract attributes "foo"
 [case testUndefinedBaseclassInNestedClass]
 class C:
     class C1(XX): pass  # E: Name "XX" is not defined
+
+[case testClassScopeImportFunction]
+class Foo:
+    from mod import foo
+
+reveal_type(Foo.foo)  # N: Revealed type is "def (x: builtins.int, y: builtins.int) -> builtins.int"
+reveal_type(Foo().foo)  # E: Invalid self argument "Foo" to attribute function "foo" with type "Callable[[int, int], int]" \
+                        # N: Revealed type is "def (y: builtins.int) -> builtins.int"
+[file mod.py]
+def foo(x: int, y: int) -> int: ...
+
+[case testClassScopeImportVariable]
+class Foo:
+    from mod import foo
+
+reveal_type(Foo.foo)  # N: Revealed type is "builtins.int"
+reveal_type(Foo().foo)  # N: Revealed type is "builtins.int"
+[file mod.py]
+foo: int
+
+[case testClassScopeImportModule]
+class Foo:
+    import mod
+
+reveal_type(Foo.mod.foo)  # N: Revealed type is "builtins.int"
```

Comment:
I think `reveal_type(Foo.mod)` will also be helpful here.

Reply:
Okay, can add!

Reply:
It's revealed as builtins.object, not types.ModuleType, but pretty sure that's just because of the stub fixtures